lib/improver/tests/argparser/test_ArgParser.py

```diff
@@ -0,0 +1,358 @@
+
+    def test_adding_single_argument_with_unexpected_length_argspec(self):
+        """Test that attempting to add an argument to the ArgParser when
+        the wrong format argspec raises an exception."""
+
+        # length of argspec is 3 - this is unexpected
+        args_to_add = [
+                        (['--foo'], 'bar', {}),
+                      ]
+
+        parser = ArgParser(central_arguments=None,
+                           specific_arguments=None)
+
+        with self.assertRaises(AttributeError):
+            parser.add_arguments(args_to_add)
+
+    def test_adding_empty_argument_list_does_nothing(self):
+        """Test that attempting to add an empty list of argspecs to the
+        ArgParser does not add any new arguments."""
+
+        args_to_add = []
+
+        # add a specific (optional) argument - ensures that even if there are
+        # no compulsory arguments, we have something...
+        # adding arguments after calling parse_args/args will do nothing, so
+        # instead create 2 instances:
+        parser1 = ArgParser(central_arguments=None,
+                            specific_arguments=[[['--optional'], {}]])
+
+        parser2 = ArgParser(central_arguments=None,
+                            specific_arguments=[[['--optional'], {}]])
+
+        parser2.add_arguments(args_to_add)
+        self.assertEqual(parser1.parse_args(), parser2.parse_args())
+
+
+class Test__args_from_selected_central_args(QuietTestCase):
+
+    """Test the _args_from_selected_args method."""
+
+    def setUp(self):
+        """Patch the ArgParser.CENTRALIZED_ARGUMENTS dictionary so test cases
+        know exactly what to expect."""
+        DUMMY_CENTRALIZED_ARGUMENTS = {'foo': (['central_foo'], {}),
+                                       'bar': (['central_bar'], {})}
+        self.patch = patch(
+                        'improver.argparser.ArgParser.CENTRALIZED_ARGUMENTS',
+                        DUMMY_CENTRALIZED_ARGUMENTS)
+        self.patch.start()
+
+    def tearDown(self):
+        """Stop the patch of ArgParser.CENTRALIZED_ARGUMENTS."""
+        self.patch.stop()
+
+    def test_retrieving_known_central_argument(self):
+        """Test that we can successfully retrieve centralized arguments from
+        the centralized argument dictionary."""
+
+        central_args_to_fetch = ('foo',)
+        expected_result = [(['central_foo'], {})]
+        self.assertEqual(ArgParser._args_from_selected_central_args(
+            central_args_to_fetch), expected_result)
+
+    def test_retrieving_unknown_central_argument(self):
+        """Test that we raise an exception when attempting to retrieve
+        centralized arguments which are not centralized argument dictionary."""
+
+        central_args_to_fetch = ('missing_central_arg',)
+        with self.assertRaises(KeyError):
+            ArgParser._args_from_selected_central_args(central_args_to_fetch)
+
+
+class Test_args(QuietTestCase):
```

Comment:
This test class is not needed.

Reply:
Probably not, because it's just an alias, but our current policy is (as far as I know) to write unit tests for all new functions/methods... I'm going to leave this unless there are any very strong objections.

Reply:
Our current policy as I understand it is to test new functionality unless it is obvious or just a wrapper for existing / external library functions: https://github.com/metoppv/improver/wiki/Code-Style-Guide#unit-testing.  So for the sake of minimising the number of lines of code, it would be best if we could get rid of this please.